apps/store/themes/store/js/list_assets.js

```diff
@@ -490,114 +490,116 @@ function resolveDomain() {
 }
 
 $(window).load(function () {
+    if (store.taxonomyAvailability) {
 // this condition will check current page is asset list page or top asset page.
-    if (window.location.href.toString().indexOf('list') > 0 || window.location.href.toString().
-            indexOf('top-assets') > 0) {
-
-        // first ajax call when page loads
-        $.ajax({
-            url: caramel.context + '/apis/taxonomies?' + resolveDomain(),
-            type: 'GET',
-            async: false,
-            headers: {
-                Accept: "application/json"
-            },
-            success: function (data) {
-                // success function will modify the REST api return data into jstree format.
-                var children;
-                try {
-                    children = Array.isArray(data[0].children);
-                } catch (e) {
+        if (window.location.href.toString().indexOf('list') > 0 || window.location.href.toString().
+                indexOf('top-assets') > 0) {
+
+            // first ajax call when page loads
+            $.ajax({
+                url: caramel.context + '/apis/taxonomies?' + resolveDomain(),
+                type: 'GET',
+                async: false,
+                headers: {
+                    Accept: "application/json"
+                },
+                success: function (data) {
+                    // success function will modify the REST api return data into jstree format.
+                    var children;
+                    try {
+                        children = Array.isArray(data[0].children);
+                    } catch (e) {
 
-                }
-
-                if (children) {
-                    data[0].id = data[0].elementName;
-                    for (var i = 0; i < data[0].children.length; i++) {
-                        data[0].children[i].id = data[0].elementName + "/" + data[0].children[i].elementName;
                     }
-                } else {
-
-                    for (var i = 0; i < (data.length); i++) {
-                        //not executng first run
-                        data[i].id = data[0].elementName + "/" + data[i].elementName;
-                    }
-                }
-                // call to the first tab creation in landing top asset page
-                createHTMLFromJsonFirst(data);
 
+                    if (children) {
+                        data[0].id = data[0].elementName;
```

Comment:
Please change this so that that the 0th element is accessed only once.

var zero = data[0] || { elementName:'',children:[] };
zero.children.forEach(function(child,index) {
     child.id = [zero.elementName , child.elementName ].join('/');
});